src/backend/vulkan/src/lib.rs

```diff
@@ -647,6 +649,285 @@ impl hal::Instance<Backend> for Instance {
             .functor
             .destroy_surface(surface.raw.handle, None);
     }
+
+    fn enumerate_available_displays<'a>(&self,adapter: &'a adapter::Adapter<Backend>)->Result<Vec<hal::display::Display<'a,Backend>>,hal::device::OutOfMemory>{
+        let display_extension = ash::extensions::khr::Display::new(&self.entry,&self.raw.inner);
+
+        let display_properties = match unsafe{display_extension.get_physical_device_display_properties(adapter.physical_device.handle)}
+        {
+            Ok(display_properties)=>display_properties,
+            Err(error)=>{
+                match error
+                {
+                    ash::vk::Result::ERROR_OUT_OF_HOST_MEMORY=>return Err(hal::device::OutOfMemory::Host.into()),
+                    ash::vk::Result::ERROR_OUT_OF_DEVICE_MEMORY=>return Err(hal::device::OutOfMemory::Device.into()),
+                    _=>panic!("Unexpected error returned")
+                };
+            }
+        };
+
+        let mut displays = Vec::new();
+        for display_property in display_properties
+        {
+            let display_handle = native::Display(display_property.display);
+
+            let supported_transforms = {
+                let mut transformations = Vec::new();
+                if display_property.supported_transforms.contains(vk::SurfaceTransformFlagsKHR::IDENTITY) {transformations.push(hal::display::SurfaceTransformation::Identity);}
```

Comment:
That is because all the functions that accept as parameter `SurfaceTransformation` (effectively there is only one) assume that only one bit will be enabled (a surface cannot be at the same time setted as Indentity or Rotated90 for example). Since there is no related error to return on that function, passing a `SurfaceTransformation` with multiple bits enabled will fall into undefined behaviour (probably the driver will use the mode related to the first enabled bit, while ignoring the rest, but i have not tested this). The same apply for the `DisplayPlaneAlpha` struct. So i thought i could avoid this problem by defining that bitflags as enums: while making the function that gather info returns the available transformations as a `Vec<SurfaceTransformation>`, the function that use `SurfaceTransformation` simply accept a enum value that can be only one of the available transformations. This was made to make the function safe by api, so that there is no need to make additions checks into the function.
If you prefer to have bitflags instead, i can replace them without problems.